mavros/src/plugins/waypoint.cpp

```diff
@@ -147,6 +147,7 @@ class WaypointPlugin : public plugin::PluginBase {
 		wp_state = WP::IDLE;
 
 		wp_nh.param("pull_after_gcs", do_pull_after_gcs, false);
+		wp_nh.param("enable_partial_push",enable_partial_push,false);
```

Comment:
Better:

```
//! @note currently (2017) only APM support partial mission update
wp_nh.param("enable_partial_push", enable_partial_push, m_uas->is_ardupilotmega());
```

Reply:
Oh, no, that would not work.

here that might be defaulted to `false` but in `connection_cb()` need to check:

```
if (!wp_nh.getParam(...)) {
// no override param - detect by FCU
enable = m_uas->is_ardupilotmega();
}
```

Reply:
Yes this is better. Then i will remove it from the APM config yaml.